ignite/handlers/ema_handler.py

```diff
@@ -125,25 +125,27 @@ def __init__(
         momentum_warmup: Optional[float] = None,
         warmup_iters: Optional[int] = None,
     ) -> None:
-        if momentum_warmup is not None and not 0 < momentum_warmup < 1:
-            raise ValueError(f"Invalid momentum_warmup: {momentum_warmup}")
         if not 0 < momentum < 1:
             raise ValueError(f"Invalid momentum: {momentum}")
-        if momentum_warmup is not None and not momentum_warmup <= momentum:
-            raise ValueError(
-                f"momentum_warmup should be less than or equal to momentum, but got "
-                f"momentum_warmup: {momentum_warmup} and momentum: {momentum}"
+        if momentum_warmup is not None:
+            warnings.warn(
+                "Argument 'momentum_warmup' will be deprecated in the future. In the current version,"
+                "it has no effect on the ema momentum. Please use the ParamStateScheduler to schedule"
+                "the ema momentum."
+            )
+        if warmup_iters is not None:
+            warnings.warn(
+                "Argument 'warmup_iters' will be deprecated in the future. In the current version, "
+                "it has no effect on the ema momentum. Please use the ParamStateScheduler to schedule"
+                "the ema momentum."
             )
-        if warmup_iters is not None and not (isinstance(warmup_iters, int) and warmup_iters > 0):
-            raise ValueError(f"Invalid warmup_iters: {warmup_iters}")
         if not isinstance(model, nn.Module):
             raise ValueError(
                 f"model should be an instance of nn.Module or its subclasses, but got"
                 f"model: {model.__class__.__name__}"
             )
-        self.momentum_warmup = momentum_warmup
+        # TODO: in the next version, rename momentum to init_momentum, which is more rigorous
```

Comment:
Maybe, we could make it as a property to fetch `getattr(engine.state, name)` ?

Reply:
@vfdev-5 You mean the current momentum? Yes, but we need to first store the initial momentum with an attribute, say `init_momentum`. Then, when we attach the handler to the engine, we know how to initialize the momentum. After that, we can then retrieve the current momentum from the engine.

A problem is that a property usually does not allow to pass the second argument in addition to `self`. It means that if we want to make it a property, we need to write it like this:
```python3
@property
def momentum(self):
    return getattr(self.engine.state, self.name)
```
So we need to store the engine and the attribute name of the engine state. This might lead to a circular reference between `Engine` and `EMAHandler`.

If you think the `getattr` too ugly. We can make simply wrap it like this:

```python3
def get_curr_momentum(self, engine, name):
    return getattr(self.engine.state, self.name)
```
Since this function is called in `_update_ema_model`, the arguments `engine` and `name` are available.

Reply:
No, you are right, probably we can not delegate everything to engine.state as we still need the initial value. I was hesitating about the renaming `momentum` -> `init_momentum`.
Let's keep it as it is and keep your comment as well. Thanks for the explanation.